chapters/3/modul-kompilujacy.tex

```diff
@@ -0,0 +1,75 @@
+
+\subsection{Obraz bazowy}
+Wybranym obrazem bazowym do kontenera zostało Ubuntu 20.04 \cite{linuxUbuntu}, jest to popularny wybór, dzięki czemu jest dobrze przetestowany i kompatybilny z wieloma narzędziami. Obrazy Ubuntu dostępne w Docker Hub to systemy oparte na Ubuntu Base, czyli lżejszej wersji Ubuntu, nieposiadającej oprogramowania, które nie jest przydatne do pracy w kontenerze takich jak interfejs graficzny lub aplikacje biurowe. Wersja 20.04 jest wersją Long-Term Support, co oznacza, że przez długi czas oferuje aktualizacje bezpieczeństwa oraz poprawki. Posiada również bogate repozytorium pakietów, oferujące wszystkie potrzebne w naszym zastosowaniu narzędzia.
+
+\subsection{Wykorzystane pakiety}
+Wszystkie pakiety konieczne do prawidłowej pracy kontenera zostały pozyskane z repozytorium Ubuntu 20.04 przy pomocy narzędzia apt-get. Obejmowały kompilator mingw, narzędzie Wine pozwalające na uruchamianie aplikacji przeznaczonych na system operacyjny Windows oraz wirtualny serwer Xvfb zapewniający wirtualne środowisko graficzne dla Wine.
+
+\subsection{Wymiana plików między kontenerem, a systemem hosta}
+Narzędzie Docker zapewnia dwie podstawowe metody wymiany plików między kontenerem, a systemem hosta - Docker Volume\cite{dockerVolume} oraz Bind mounts\cite{dockerBindMounts}. Docker Volume jest mechanizmem zarządzanym przez system Docker, odpowiada on za tworzenie, zarządzanie i usuwanie danych. Możliwe jest współdzielenie jednego wolumenu przez liczne kontenery. Bind mounts to prostsze rozwiązanie polegające na bezpośrednim połączeniu wybranego katalogu w systemie hosta z katalogiem kontenera. Zapewnia to większą elastyczność, ponieważ nie wymaga tworzenia osobnego wolumenu, a jedynie powiązania ścieżki bezwzględnej na systemie hosta, ze ścieżką wewnątrz kontenera, jednak tym samym zmniejsza to izolację kontenera od systemu. W naszym rozwiązaniu zostały wykorzystane bind mounts, ponieważ każde sprawdzane rozwiązanie otrzymuje swój osobny folder powiązany z jego id, który wiążemy z osobnym, nowo utworzonym kontenerem. Eliminuje to potrzebę zarządzania licznymi wolumenami lub dzielenia wolumenu między kilka kontenerów co z kolei zmniejszyłoby izolację między zadaniami. Aby zaimplementowany kontener poprawnie skompilował i uruchomił program, struktura powiązanego folderu powinna wyglądać w następujący sposób:
```

Comment:
> wymiany plików między kontenerem, a systemem hosta

wymiany plików między kontenerem a systemem hosta (bez przecinka, bo to jest spójnik, a nie przeciwstawienie)

Reply:
Zmieniłem system hosta we wszystkich miejscach na "system macierzysty"